### DISCUSSION TITLE
Unwanted openlayers map added to my app

### DISCUSSION THREAD

**Discussion by mlandsfeld** [tethysplatform/tethys#735] [Q&A] [answered]
I have two OpenLayers maps, side by side, in my application and a few days ago a third map appeared and I have not been able to remove it. I think that what happened was that I was resetting the Spatial Dataset Service in the Tethys portal and the page seemed to refresh suddenly. When I went back to the app, I had a third map. It is a fully functional map but it's not displaying any layers other than the default background map. There has not been any code added to my home.html where the other two maps are defined. Does anyone know how I can purge this wayward map?

**Comment by swainn** [comment]
Hey @mlandsfeld that sounds perplexing. It'll be easier to get to the bottom of this by looking at the code. Can you please share the code for your `home.html` and the `home` controller function?

**Reply by mlandsfeld** [reply]
You can download those two files here:
https://data.chc.ucsb.edu/people/marty/tethys/ml_tethys_code.zip
Thanks

**Comment by swainn** [comment, marked as answer]
@mlandsfeld so sorry for the delay on this. I'm not seeing anything obvious in the code you shared. A few things to try:

* Try removing line 44 entirely: `{# % gizmo eo_map_options % #}`  It should be commented out, but maybe having the `%` signs breaks the commenting out effect?
* Make the id/target of the second map more unique. Both map divs have a class called `map` so I wonder if that is causing confusion for OpenLayers here.

```
<div id="forecast_map" class="map"></div>
```

```
var map = new ol.Map({
    target: 'forecast_map',
```

I doubt either of those is the problem, however. Could you also share the main.js? Or if you have the project on GitHub, a link to the repository?

**Reply by mlandsfeld** [reply]
Good eye Nathan! Renaming the "map" class worked. It is so much nicer to develop with now... Thanks!!

**Reply by swainn** [reply]
Excellent. I'm glad it worked.
